front/src/modules/timesStops/hooks/useOutputTableData.ts

```diff
@@ -19,48 +19,52 @@ import { ARRIVAL_TIME_ACCEPTABLE_ERROR } from '../consts';
 import { computeInputDatetimes } from '../helpers/arrivalTime';
 import computeMargins, { getTheoreticalMargins } from '../helpers/computeMargins';
 import { formatSchedule } from '../helpers/scheduleData';
+import { matchOpRefAndOp, getTrackReferenceLabel, getOperationalPointName } from '../helpers/utils';
 import { type ScheduleEntry, type TimesStopsRow } from '../types';
 
 const useOutputTableData = (
-  simulatedTrain?: SimulationResponseSuccess['final_output'],
-  simulationSummary?: SimulationSummary,
-  operationalPoints?: PathPropertiesFormatted['operationalPoints'],
   selectedTrain?: Train,
-  path?: PathfindingResultSuccess
+  isValid?: boolean,
+  simulatedTrain?: SimulationResponseSuccess['final_output'],
+  simulatedPath?: PathfindingResultSuccess,
+  simulatedPathItemTimes?: Extract<SimulationSummary, { isValid: true }>['pathItemTimes'],
+  operationalPoints?: PathPropertiesFormatted['operationalPoints']
 ): TimesStopsRow[] => {
-  const { t } = useTranslation();
+  const { t } = useTranslation('operational-studies');
   const { getTrackSectionsByIds } = useScenarioContext();
 
   const [rows, setRows] = useState<TimesStopsRow[]>([]);
 
+  // Extract common properties between valid and invalid trains
   const scheduleByAt: Record<string, ScheduleEntry> = keyBy(selectedTrain?.schedule, 'at');
   const theoreticalMargins = selectedTrain && getTheoreticalMargins(selectedTrain);
 
+  // Format input path step rows
   const pathStepRows = useMemo(() => {
-    if (!selectedTrain || !simulationSummary?.isValid || !path) return [];
-
-    const { pathItemTimes } = simulationSummary;
+    if (!selectedTrain || !operationalPoints) return undefined;
 
     const startDatetime = new Date(selectedTrain.start_time);
     let lastReferenceDate = startDatetime;
-
+    let mapWaypointCount = 0;
     return selectedTrain.path.map((pathStep, index) => {
-      const schedule: ScheduleEntry | undefined = scheduleByAt[pathStep.id];
+      const opPositionOnPath = simulatedPath?.path_item_positions[index];
+      const matchingOperationalPoint = operationalPoints.find((op) =>
+        opPositionOnPath ? op.position === opPositionOnPath : matchOpRefAndOp(op, pathStep)
+      );
 
-      const computedArrival = new Date(startDatetime.getTime() + pathItemTimes.final[index]);
+      if ('track' in pathStep) mapWaypointCount += 1;
+      const name = isValid
+        ? matchingOperationalPoint?.extensions?.identifier?.name
+        : getOperationalPointName(matchingOperationalPoint, pathStep, mapWaypointCount, t);
```

Comment:
We can also find the name in `getOperationalPointName`
```suggestion
      const name = getOperationalPointName(matchingOperationalPoint, pathStep, mapWaypointCount, t);
```

Reply:
Done!